File: typescript/src/refactoring/parallel-change/sprout-change/sprout-change.test.ts
import {describe, expect, it} from 'vitest'
import {calculateTotal, CartItem, Region} from './sprout-change'

describe('Sprout Change', () => {
  function executeSubject(cart: CartItem[], region: Region) {
    return calculateTotal(cart, region)
  }

  const cart: CartItem[] = [
    {id: 'p1', price: 10, qty: 2, category: 'general'},
    {id: 'b1', price: 20, qty: 1, category: 'books'},
    {id: 'f1', price: 15, qty: 4, category: 'food'},
  ]

  it('should calculate the total for EU', () => {
    expect(executeSubject(cart, 'EU')).toEqual(104)
  })

  it('should calculate the total for US', () => {
    expect(executeSubject(cart, 'US')).toEqual(107)
  })
})

File: go/refactoring/parallel_change/sprout_change/sprout_change_test.go
package sprout_change

import "testing"

var cart = []CartItem{
	{ID: "p1", Price: 10, Qty: 2, Category: "general"},
	{ID: "b1", Price: 20, Qty: 1, Category: "books"},
	{ID: "f1", Price: 15, Qty: 4, Category: "food"},
}

func executeSubject(cart []CartItem, region Region) float64 {
	return CalculateTotal(cart, region)
}

func TestShouldCalculateTotalForEU(t *testing.T) {
	result := executeSubject(cart, RegionEU)
	if result != 104 {
		t.Errorf("expected 104, got %v", result)
	}
}

func TestShouldCalculateTotalForUS(t *testing.T) {
	result := executeSubject(cart, RegionUS)
	if result != 107 {
		t.Errorf("expected 107, got %v", result)
	}
}
